stripe/src/main/java/com/stripe/android/view/CardNumberEditText.kt

```diff
@@ -206,9 +236,25 @@ class CardNumberEditText @JvmOverloads constructor(
     }
 
     @JvmSynthetic
-    internal fun updateCardBrandFromNumber(bin: Bin?) {
-        cardBrand = bin?.let {
-            CardUtils.getPossibleCardBrand(it)
-        } ?: CardBrand.Unknown
+    internal fun updateCardBrand(inputBin: Bin?) {
+        // cancel in-flight job
+        accountRangeRepositoryjob?.cancel()
+        accountRangeRepositoryjob = null
+
+        accountRangeRepositoryjob = CoroutineScope(workDispatcher).launch {
+            if (inputBin != null) {
+                onAccountRangeResult(
+                    cardAccountRangeRepository.getAccountRange(inputBin.value)
+                )
+            } else {
+                onAccountRangeResult(null)
```

Comment:
Will this view method reference leak the view? I think the job should be cancelled and restarted as part of the view's lifecycle. Are we able to add a test for this?

Reply:
Good catch. Added this logic in `onDetachedFromWindow()` and added a test.

Reply:
Cool, and then the `TextWatcher` will start the job again, right?